ios/brave-ios/Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift

```diff
@@ -75,6 +75,69 @@ public struct AssetGroupViewModel: WalletAssetGroupViewModel, Identifiable, Equa
       return partialResult + assetValue
     }
   }
+
+  /// Sort by the group's total fiat/value
+  static func sorted(
+    lhs: AssetGroupViewModel,
+    rhs: AssetGroupViewModel
+  ) -> Bool {
+    if lhs.totalFiatValue == rhs.totalFiatValue {
+      return sameBalanceSort(lhs: lhs, rhs: rhs)
+    } else {
+      return lhs.totalFiatValue > rhs.totalFiatValue
+    }
+  }
+
+  /// Sorts primary networks to be first (Solana Mainnet first primary network),  then sorts alphabetically.
+  /// Used when two tokens have the same balance or fiat value (typically 0 / $0).
+  private static func sameBalanceSort(lhs: AssetGroupViewModel, rhs: AssetGroupViewModel) -> Bool {
+    if case .account(let lhsAccount) = lhs.groupType, case .account(let rhsAccount) = rhs.groupType
+    {
+      if lhsAccount.coin == .fil && rhsAccount.coin == .fil {
+        if lhsAccount.keyringId == .filecoin && rhsAccount.keyringId != .filecoin {
+          return true
+        } else if lhsAccount.keyringId != .filecoin && rhsAccount.keyringId == .filecoin {
+          return false
+        }
+      } else if lhsAccount.coin == .btc && rhsAccount.coin == .btc {
+        if lhsAccount.keyringId == .bitcoin84 && rhsAccount.keyringId != .bitcoin84 {
+          return true
+        } else if lhsAccount.keyringId != .bitcoin84 && rhsAccount.keyringId == .bitcoin84 {
+          return false
+        }
+      } else {
+        if lhsAccount.keyringId == .solana && rhsAccount.keyringId != .solana {
+          return true
+        } else if lhsAccount.keyringId != .solana && rhsAccount.keyringId == .solana {
+          return false
+        }
+      }
+    }
+    if case .network(let lhsNetwork) = lhs.groupType, case .network(let rhsNetwork) = rhs.groupType
+    {
+      let isLHSPrimaryNetwork = WalletConstants.primaryNetworkChainIds.contains(lhsNetwork.chainId)
+      let isRHSPrimaryNetwork = WalletConstants.primaryNetworkChainIds.contains(rhsNetwork.chainId)
+      if isLHSPrimaryNetwork && !isRHSPrimaryNetwork {
+        return true
+      } else if !isLHSPrimaryNetwork && isRHSPrimaryNetwork {
+        return false
+      } else if isLHSPrimaryNetwork, isRHSPrimaryNetwork,
+        lhsNetwork.chainId != rhsNetwork.chainId,
+        lhsNetwork.chainId == BraveWallet.SolanaMainnet
+      {
+        // Solana Mainnet to be first primary network
+        return true
+      } else if isLHSPrimaryNetwork, isRHSPrimaryNetwork,
+        lhsNetwork.chainId != rhsNetwork.chainId,
+        rhsNetwork.chainId == BraveWallet.SolanaMainnet
+      {
+        // Solana Mainnet to be first primary network
+        return false
+      }
+      return lhs.id < rhs.id
+    }
+    return lhs.id < rhs.id
+  }
```

Comment:
Is it possible to sort this as an array of tuples?
https://holyswift.app/using-tuples-to-complex-sorting-operations-in-swift/
```
let sortedMovieList = movieList.sorted {
    ($0.title, $0.year, $0.director) < ($1.title, $1.year, $1.director)
}

for movie in sortedMovieList {
    print(movie)
}
```

Reply:
this requires each property inside the tuple conform `Comparable` protocol. In your example, they are either String or Int, they are all `Comparable` using the default  `<` implementation. In our case, we have more complex logic of comparison of lhs and rhs object rather than the default `<` implementation.

Reply:
It doesn't feel right if swift's sugar doesn't support that, neverthelss if tuple sort could not be applied here then at least this algorithm needs a unit test.